app/controllers/saml_idp_controller.rb

```diff
@@ -33,6 +33,12 @@ def logout
     return sign_out_with_flash if raw_saml_request.nil?
 
     decode_request(raw_saml_request)
+
+    # Plumb the fingerprint through to the internal service_provider representation
+    if saml_request && matching_cert
+      saml_request.service_provider.fingerprint = Fingerprinter.fingerprint_cert(matching_cert)
+    end
+
     track_logout_event
```

Comment:
quick heads up we have an error in INT that has to do with this and a nil cert fingerprint, I need to do some digging to make sure it won't be an issue if we ship this to prod

Reply:
https://github.com/18F/identity-idp/pull/4890